src/lib/index.js

```diff
@@ -160,6 +162,22 @@ module.exports = function npmRunAll(
             prefixOptions.push(...toOverwriteOptions(packageConfig));
         }
 
+        for (let i = 0, len = patterns.length; i < len; ++i) {
+            patterns[i] = patterns[i].replace(/[{]([*@]|\d+)[}]/g, (match, index) => {
+                if (index === "@") {
+                    return shellQuote.quote(rest.slice(1));
+                }
+                if (index === "*") {
+                    return shellQuote.quote([rest.slice(1).join(" ")]);
+                }
+                const position = parseInt(index, 10);
+                if (position >= 0 && position < rest.length) {
```

Comment:
And in the current specification, `{0}` is always `--`.
Only a few, similar to a `$0` of shell script.